### PR TITLE
doc: add EDA and data validation sections with other minor updates

### PR DESCRIPTION
**Pull request by thurstonchen**
# Type of PR

- Documentation changes

## Purpose

Address below review comments for the V2 merge.
- Added descriptions for each imported notebooks.
- Added a section for the data validation notebook.
- Added a section for the EDA notebook.
- Added .gif files for Fabric 'Environment' setup section.
- Removed not required .png screenshots.

## Does this introduce a breaking change? If yes, details on what can break
NO

## Author pre-publish checklist

- [ ] Added test to prove my fix is effective or new feature works
- [ ] No PII in logs
- [ ] Made corresponding changes to the documentation

## Validation steps

N/A

## Issues Closed or Referenced
N/A

### CODE REVIEW

File: single_tech_samples/fabric/feature_engineering_on_fabric/README.md [hunk 1]
```diff
@@ -114,14 +120,30 @@ As described above, the sample uses Microsoft Fabric as the data analytics platf
...
+   | exploratory_data_analysis | Explore and analyze data in `Staging Zone` to discover patterns, spot anomalies, test a hypothesis, or check assumptions by data scientists according to data analyzing or machine learning model training requirements |
+   | data_validation | Validate data in `Staging Zone` by user-defined rules using [Great Expectations](https://docs.greatexpectations.io/docs/) |
+   | data_cleansing | Cleanse data in `Staging Zone` and sink them to `Standardized Zone` |
+   | data_transformation | Transform cleansed data and sink them to `Standardized Zone` |
+   | feature_set_registration | Create features based on transformed data and register them to Azure ML managed feature store |
+   | data_catalog_and_lineage | Utility functions for registering data assets and lineage to Microsoft Purview |
+   | model_training | Train ML models using features values retrieved from Azure ML managed feature store |
+   | model_inferencing | Do some inferencing based on trained model |
+   | feature_set_retrieval | Utility functions for retrieving features values for model training |
+   | utils | The common utility functions |
```
Comment (promisinganuj) [inline single_tech_samples/fabric/feature_engineering_on_fabric/README.md:145]:
```suggestion
   | data_ingestion | Ingest data from the `Landing Zone` to the `Staging Zone`. |
   | exploratory_data_analysis | Explore and analyze data in `Staging Zone` to discover patterns, spot anomalies, test a hypothesis, or check assumptions by data scientists according to data analyzing or ML model training requirements. |
   | data_validation | Validate data in `Staging Zone` by user-defined rules using [Great Expectations](https://docs.greatexpectations.io/docs/) library. |
   | data_cleansing | Cleanse data in `Staging Zone` and sink it to `Standardized Zone`. |
   | data_transformation | Transform cleansed data and sink it to `Standardized Zone`. |
   | feature_set_registration | Create features based on transformed data and register them to Azure ML managed feature store. |
   | data_catalog_and_lineage | Utility functions for registering data assets and lineage to Microsoft Purview. |
   | model_training | Train ML models using features values retrieved from Azure ML managed feature store. |
   | model_inferencing | Perform inferencing based on trained model. |
   | feature_set_retrieval | Utility functions for retrieving features values for model training. |
   | utils | Common utility functions. |
```

File: single_tech_samples/fabric/feature_engineering_on_fabric/README.md [hunk 2]
```diff
@@ -379,6 +400,40 @@ If the pipeline executes successfully, you can verify the features registered in
 
   ![feature_lineage](./images/managed_feature_store.gif)
 
+### Exploratory data analysis (EDA)
```
Comment (promisinganuj) [inline single_tech_samples/fabric/feature_engineering_on_fabric/README.md:403]: Few comments:
- Suggest moving it to as an H2 header after "Model training and inferencing" as below:

## Model training and inferencing

## Additional (optional) Features

### Exploratory data analysis (no need to put the short form 'EDA' here)

### Data Validations

File: single_tech_samples/fabric/feature_engineering_on_fabric/README.md [hunk 3]
```diff
@@ -30,6 +30,8 @@ This sample focuses on constructing a feature engineering system using Azure ML
     - [Data lineage](#data-lineage)
     - [Feature lineage](#feature-lineage)
   - [Verify the features in Feature Store](#verify-the-features-in-feature-store)
+  - [Exploratory data analysis (EDA)](#exploratory-data-analysis-eda)
+  - [Data validation](#data-validation)
 - [Model training and inferencing](#model-training-and-inferencing)
   - [Model training](#model-training)
   - [Model inferencing](#model-inferencing)
```
Comment (promisinganuj) [inline single_tech_samples/fabric/feature_engineering_on_fabric/README.md:37]: Update line 53:

In addition to the main flow, there are optional steps for performing 'exploratory data analysis' and 'data validations' (illustrated by dotted lines in the diagram). These features are not part of the main processing steps, but are covered separately under [Additional Features](#additional-features).

File: single_tech_samples/fabric/feature_engineering_on_fabric/README.md [hunk 4]
```diff
@@ -379,6 +400,40 @@ If the pipeline executes successfully, you can verify the features registered in
 
   ![feature_lineage](./images/managed_feature_store.gif)
 
+### Exploratory data analysis (EDA)
+
```
Comment (promisinganuj) [inline single_tech_samples/fabric/feature_engineering_on_fabric/README.md:404]: Ordering of the content. Suggest, first giving a brief overview of what EDA is and then add the notebook specific detail.

Also, provide link to the notebook at-least once. i.e, 

replace `exploratory_data_analysis` with [exploratory_data_analysis.ipynb](./src/notebooks/exploratory_data_analysis.ipynb)

And do the same for data validation step.

File: single_tech_samples/fabric/feature_engineering_on_fabric/README.md [hunk 5]
```diff
@@ -379,6 +400,40 @@ If the pipeline executes successfully, you can verify the features registered in
...
+```python
+# Report Data Quality Metrics to Azure Monitor using python Azure Monitor open-census exporter 
+import logging
+import time
+from opencensus.ext.azure.log_exporter import AzureLogHandler
+
+AZURE_MONITOR_SECRET = "InstrumentationKey=xxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxx"
+```
+
+For more details, please refer to this article, [Data Validation with Great Expectations on Microsoft Fabric](https://medium.com/@756025472/enhancing-data-validation-with-great-expectations-and-sending-logs-to-azure-monitor-19ba7d345d20).
```
Comment (promisinganuj) [inline single_tech_samples/fabric/feature_engineering_on_fabric/README.md:435]:
```suggestion
For more details, refer to the article, [Data validation with Great Expectations on Microsoft Fabric](https://medium.com/@756025472/enhancing-data-validation-with-great-expectations-and-sending-logs-to-azure-monitor-19ba7d345d20).
```

File: single_tech_samples/fabric/feature_engineering_on_fabric/README.md [hunk 6]
```diff
@@ -114,14 +120,30 @@ As described above, the sample uses Microsoft Fabric as the data analytics platf
...
 
 3. Import the sample notebooks
 
-   After the lakehouse is created, go back to the workspace. Click the 'New' button again and select 'Import notebook' from the menu. Navigate to `src/notebooks` folder under the repo, and select all the notebooks to import:
+   After the lakehouse is created, go back to the workspace. Click the 'New' button from the **Data Engineering** or the **Data Science** homepage and select 'Import notebook' from the menu. Navigate to `src/notebooks` folder under the repo, and select all the notebooks to import:
 
    ![import all notebooks](./images/import_all_notebooks.png)
 
+   And below are brief introductions for all notebooks.
+
```
Comment (promisinganuj) [inline single_tech_samples/fabric/feature_engineering_on_fabric/README.md:132]: I suggest adding links to the notebooks in the table.

**Comment by promisinganuj** [comment]
Hi @thurstonchen , @siliang-j-1225 , @cchenshu , great work. I have added few minor comments. Please have a look.

File: single_tech_samples/fabric/feature_engineering_on_fabric/README.md (hunk 3 above)
Comment (thurstonchen) [reply single_tech_samples/fabric/feature_engineering_on_fabric/README.md:37]: OH! Forgot to update this part accordingly, thanks for the reminder!

File: single_tech_samples/fabric/feature_engineering_on_fabric/README.md (hunk 4 above)
Comment (thurstonchen) [reply single_tech_samples/fabric/feature_engineering_on_fabric/README.md]: Hi @promisinganuj, 
Thanks for your great suggestions as always! 
Made some updates to address several of them, thanks for your help to check again! :- )

File: single_tech_samples/fabric/feature_engineering_on_fabric/README.md (hunk 6 above)
Comment (thurstonchen) [reply single_tech_samples/fabric/feature_engineering_on_fabric/README.md:133]: Done! :- )

**Review by promisinganuj — APPROVED** [review]
This looks perfect, well done!

**Pull request merged**